### ISSUE TITLE
Delete page history

### ISSUE THREAD

**Issue by RobeeeJay** [justmiles/node-atlassian-confluence#5]
I know the command line tools are capable of deleting page history, but is there any way to do it with this module? I couldn't find anything in the REST API docs so I'm not actually sure it's even possible. :/

**Comment by justmiles** [comment]
I'm not sure. I could definitely use it as well. I'll take a look at the
API docs. If nothing else, maybe we can delete specific versions and go
from there.

**Comment by RobeeeJay** [comment]
After a few hours of playing around, I've so far not managed to work out a way of doing it very efficiently. Here is what I've learnt:

There doesn't seem to be any way to get a total page history, all you can do is get a link to the previously updated page, and even then this is just a link not any useful isolated ID property.

This link appears to be a child page, accessible only through the child REST endpoints. There is no way to delete this child page via the REST API, as the child endpoints don't have a DELETE method and the main content endpoint shows a 404 when you try to delete a child page with it.

So, as far as I can tell, the REST API is useless for deleting versions, however... I did manage to work out a way of deleting a version of a page if you know it's ID, which you can get from the REST API (even if it is just the last version).

Basically you have to POST to the same FORM endpoint the confluence page does when you check the box and hit delete on the versions page. It's not pretty, and you basically have to run a very long tedious loop of pulling the main page to get the previous version, delete that version, and repeat.

I've attached a rough script to achieve that, note that the confluence username/password and host domain are provided by environmental variables.

[confluence-history.js.txt](https://github.com/justmiles/node-atlassian-confluence/files/176336/confluence-history.js.txt)
